.github/workflows/python.yml

```diff
@@ -7,53 +7,25 @@ on:
       - 'build/python-client/**'
       - 'pkg/api/*.proto'
       - '.github/workflows/python.yml'
-
+      - 'docs/python_armada_client.md'
 
 jobs:
   run-tox:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      # Generate the proto files for python
-      - run: make python
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.9' # Version range or exact version of a Python version to use, using SemVer's version range syntax
-      # Use https://github.com/marketplace/actions/install-poetry-action for setting up poetry
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-
-      #----------------------------------------------
-      #       load cached venv if cache exists
-      #----------------------------------------------
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v2
-        with:
-          path: .venv
-          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
-      #----------------------------------------------
-      # install dependencies if cache does not exist
-      #----------------------------------------------
-      - name: Install dependencies
-        working-directory: client/python
-        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-        run: poetry install --no-interaction --no-root
-      #----------------------------------------------
-      # install your root project, if required
-      #----------------------------------------------
-      - name: Install library
-        working-directory: client/python
-
-        run: poetry install --no-interaction
-      #----------------------------------------------
+          python-version: '3.9'
+      - run: pip install tox
```

Comment:
I'd suggest always using a specific version.
To avoid new versions potentially breaking the build.

Reply:
I can do that if you feel strongly; but `tox` is a tool used for coordinating CI runs moreso than something we actually depend on. I've been using it for nearly a decade and have never seen a release that broke things (without loud messages about deprecation in advance). For that reason, I generally prefer leaving it unpinned (so the deprecation warnings are surfaced when we still have months to fix them).

Reply:
All right. In that case I agree.